python_modules/libraries/dagster-airflow/dagster_airflow/resources/airflow_persistent_db.py

```diff
@@ -0,0 +1,179 @@
+import datetime
+import importlib
+import os
+import tempfile
+from typing import List, Mapping, Optional
+
+import airflow
+import pytz
+from airflow.models.connection import Connection
+from airflow.models.dag import DAG
+from airflow.models.dagrun import DagRun
+from airflow.utils import db
+from dagster import (
+    Array,
+    DagsterInvariantViolationError,
+    DagsterRun,
+    Field,
+    InitResourceContext,
+    ResourceDefinition,
+    _check as check,
+)
+from dagster._core.instance import AIRFLOW_EXECUTION_DATE_STR
+from dateutil.parser import parse
+
+from dagster_airflow.utils import (
+    Locker,
+    create_airflow_connections,
+    is_airflow_2_loaded_in_environment,
+    serialize_connections,
+)
+
+# pylint: disable=no-name-in-module,import-error
+if is_airflow_2_loaded_in_environment():
+    from airflow.utils.state import DagRunState
+    from airflow.utils.types import DagRunType
+else:
+    from airflow.utils.state import State
+
+    # pylint: enable=no-name-in-module,import-error
+
+
+class AirflowPersistentDatabase:
+    """
+    A persistent Airflow database Dagster resource.
+
+    """
+
+    def __init__(self, airflow_home_path: str, dagster_run: DagsterRun, uri: str):
+        self.airflow_home_path = airflow_home_path
+        self.dagster_run = dagster_run
+        self.uri = uri
+
+    @staticmethod
+    def from_resource_context(context: InitResourceContext) -> "AirflowPersistentDatabase":
+        airflow_home_path = os.path.join(tempfile.gettempdir(), f"dagster_airflow_{context.run_id}")
+        uri = context.resource_config["uri"]
+        os.environ["AIRFLOW__DATABASE__SQL_ALCHEMY_CONN"] = uri
+        os.makedirs(airflow_home_path, exist_ok=True)
+        with Locker(airflow_home_path):
+            # reload airflow to ensure new db connection is used
+            if is_airflow_2_loaded_in_environment():
+                importlib.reload(airflow.configuration)
+                importlib.reload(airflow.settings)
+                importlib.reload(airflow)
+            else:
+                importlib.reload(airflow)
+            # make sure connections are created
+            create_airflow_connections(
+                [Connection(**c) for c in context.resource_config["connections"]]
+            )
```

Comment:
I seem to recall very similar code in the `AirflowEphemeralDatabase`. Can these be consolidated?